src/output.cpp

```diff
@@ -345,5 +345,67 @@ namespace mamba
     {
         return Context::instance().quiet || Context::instance().json || Context::instance().no_progress_bars;
     }
-}
 
+    void Console::json_write(const std::string& key, bool value)
+    {
+        json_write(key, value ? std::string("true") : std::string("false"));
+    }
+
+    void Console::json_write(const std::string& key, const std::string& value)
+    {
+        if (Context::instance().json)
+            json_log[json_hier + '/' + key] = value;
+    }
+
+    void Console::json_write(const nlohmann::json& j)
+    {
+        if (Context::instance().json)
+        {
+            nlohmann::json tmp = j.flatten();
+            for (auto it = tmp.begin(); it != tmp.end(); ++it)
+                json_log[json_hier + it.key()] = it.value();
+        }
+    }
+
+    void Console::json_append(const std::string& value)
+    {
+        if (Context::instance().json)
+        {
+            json_log[json_hier + '/' + std::to_string(json_index)] = value;
+            json_index += 1;
+        }
+    }
+
+    void Console::json_append(const nlohmann::json& j)
+    {
+        if (Context::instance().json)
+        {
+            nlohmann::json tmp = j.flatten();
+            for (auto it = tmp.begin(); it != tmp.end(); ++it)
+                json_log[json_hier + '/' + std::to_string(json_index) + it.key()] = it.value();
+            json_index += 1;
+        }
+    }
+
+    void Console::json_down(const std::string& key)
+    {
+        if (Context::instance().json)
+        {
+            json_hier += '/' + key;
+            json_index = 0;
+        }
+    }
+
+    void Console::json_up()
+    {
+        if (Context::instance().json)
+            json_hier.erase(json_hier.rfind('/'));
+    }
+
+    void Console::json_print()
+    {
+        if (Context::instance().json)
+            std::cout << std::setw(4) << Console::instance().json_log.unflatten() << std::endl;
+    }
```

Comment:
This should acquire the mutex, which strengthens my feeling that this (and all the json related methods) should be in a dedicated class.

Reply:
:+1: